server/src/main/java/com/cloud/projects/ProjectManagerImpl.java

```diff
@@ -313,7 +338,29 @@ public boolean deleteProject(long projectId) {
         CallContext.current().setProject(project);
         _accountMgr.checkAccess(ctx.getCallingAccount(), AccessType.ModifyProject, true, _accountMgr.getAccount(project.getProjectAccountId()));
 
-        return deleteProject(ctx.getCallingAccount(), ctx.getCallingUserId(), project);
+        if (isCleanup != null && isCleanup) {
+            return deleteProject(ctx.getCallingAccount(), ctx.getCallingUserId(), project);
+        } else {
+            List<VMTemplateVO> userTemplates = _templateDao.listByAccountId(project.getProjectAccountId());
+            List<VMSnapshotVO> vmSnapshots = _vmSnapshotDao.listByAccountId(project.getProjectAccountId());
+            List<UserVmVO> vms = _userVmDao.listByAccountId(project.getProjectAccountId());
+            List<VolumeVO> volumes = _volumeDao.findDetachedByAccount(project.getProjectAccountId());
+            List<NetworkVO> networks = _networkDao.listByOwner(project.getProjectAccountId());
+            List<? extends Vpc> vpcs = _vpcMgr.getVpcsForAccount(project.getProjectAccountId());
+
+            Optional<String> message = Stream.of(userTemplates, vmSnapshots, vms, volumes, networks, vpcs)
```

Comment:
Overall, LGTM. But would it be better, if we could inform the user about all the resources that are tied to the project as opposed to the first one that's found - by doing something like this:
```
 List<String> message = Stream.of(userTemplates, vmSnapshots, vms, volumes, networks, vpcs)
                    .filter(entity -> !entity.isEmpty())
                    .map(entity -> entity.size() + " " +  entity.get(0).getEntityType().getSimpleName())
                    .collect(Collectors.toList());
 ...                   
 String msg = String.join(",", message) + " to clean up";
 CloudRuntimeException e = new CloudRuntimeException("Can't delete the project yet because it has " + msg);
 ...
  ```                  
  Such that the end error would look like:
  


![image](https://user-images.githubusercontent.com/10495417/105985969-3ee44000-60c2-11eb-9438-e829fbcb7e04.png)

Or is this too much info??

Reply:
@Pearl1594 I can do that